### ISSUE TITLE
Automatic http support for `foo_config` settings 

### ISSUE THREAD

**Issue by jayvdb** [coala/coala#4072]
There are a lot of bears which have a `x_config` setting, such as `cmakelint_config`, `eslint_config`, `jshint_config`, `rubocop_config`, `scalalint_config`, `tailor_config`, `tslint_config`.

`checkstyle_configs` is the only similar one which has other strange rules to it.

That should become a reserved suffix, and coala core should provide automatic checking that the file exists, and allow http with local caching.  Note a basic local caching algorithm can be simply to pin everything in the cache until the cache is cleared with the existing command line options, and then the server and `.coafile` can always override this by adding version numbers to the filenames of the config files.

We could go one step further in coala core, and have a global setting for base location where those config files can be found, which could be a local directory, git repo, or http prefix, and they are auto fetched & used if present.  At least for git, this would reduce the number of pings needed to check that the local cache is up to date.

**Comment by satwikkansal** [comment]
So the value for these `_config` type settings of some selected bears can either be a "local path" or a "git repo url" or a "http prefix" and coala should be able to handle it?

And once the config type file is fetched it is cached by coala until cache is flushed. If the cache is flushed, we fetch it again from the url provided in the value of the setting in the `.coafile`.

**Comment by jayvdb** [comment]
yup.  As this would impact the bears, I suggest you build on top of be72730 (c.f. https://github.com/coala/coala/issues/4115), and then rebase on top of master (and maybe #4115 is ready by then)
